internal/gatewayapi/route.go

```diff
@@ -553,6 +562,28 @@ func (t *Translator) ProcessTLSRoutes(tlsRoutes []*v1alpha2.TLSRoute, gateways [
 	return relevantTLSRoutes
 }
 
+func getUpstreamConfig(obj client.Object) ir.UpstreamConfig {
```

Comment:
It's ambiguous whether these annotations are TLSRoute-specific. The function and annotation names are generic, but the comment on the first annotation says TLSRoute.

If they could apply to all routes, but only apply to TLSRoute for now, maybe `obj` -> `route` here, rename annotation to `AnnotationRouteX`, and then be explicit in the comment that only `TLSRoute` is currently supported?

Reply:
You're right it can be generic but I only wired up TLSRoute for now. Updated comments and vars in latest commits.